src/plots/waterfallPlot.ts

```diff
@@ -0,0 +1,195 @@
+     * @returns {Waterfall} The calling Waterfall Plot.
+     */
+    public connectorsEnabled(enabled: boolean): Waterfall<X, Y>;
+    public connectorsEnabled(enabled?: boolean): any {
+      if (enabled === undefined) {
+        return this._connectorsEnabled;
+      }
+      this._connectorsEnabled = enabled;
+      return this;
+    }
+
+    /**
+     * Gets the AccessorScaleBinding for whether a bar represents a total or a delta.
+     */
+    public total<T>(): Plots.AccessorScaleBinding<T, boolean>;
+    /**
+     * Sets total to a constant number or the result of an Accessor
+     * 
+     * @param {Accessor<boolean>}
+     * @returns {Waterfall} The calling Waterfall Plot.
+     */
+    public total(total: Accessor<boolean>): Waterfall<X, Y>;
+    public total<T>(total?: Accessor<boolean> | T | Accessor<T>): any {
+      if (total === undefined) {
+        return this._propertyBindings.get(Waterfall._TOTAL_KEY);
+      }
+      this._bindProperty(Waterfall._TOTAL_KEY, total, null);
+      return this;
+    }
+
+    protected _additionalPaint(time: number) {
+      this._connectorArea.selectAll("line").remove();
+      if (this._connectorsEnabled) {
+        Utils.Window.setTimeout(() => this._drawConnectors(), time);
+      }
+    }
+
+    protected _createNodesForDataset(dataset: Dataset) {
+      var drawer = super._createNodesForDataset(dataset);
+      this._connectorArea = this._renderArea.append("g").classed(Waterfall._CONNECTOR_AREA_CLASS, true);
+      return drawer;
+    }
+
+    protected _extentsForProperty(attr: string) {
+      var primaryAttr = "y";
+      if (attr === primaryAttr) {
+        return [this._extent];
+      } else {
+        return super._extentsForProperty(attr);
+      }
+    }
+
+    protected _generateAttrToProjector() {
+      var attrToProjector = super._generateAttrToProjector();
+
+      var yScale = this.y().scale;
+      var totalAccessor = Plot._scaledAccessor(this.total());
+
+      var yAttr = this.attr("y");
+      if (yAttr === undefined) {
+        attrToProjector["y"] = (d, i, dataset) => {
+          var isTotal = totalAccessor(d, i, dataset);
+          if (isTotal) {
+            return Plot._scaledAccessor(this.y())(d, i, dataset);
+          } else {
+            var currentSubtotal = this._subtotals[i];
+            var priorSubtotal = this._subtotals[i - 1];
+            if (currentSubtotal > priorSubtotal) {
+              return yScale.scale(<any> currentSubtotal);
+            } else {
+              return yScale.scale(<any> priorSubtotal);
+            }
+          }
+        };
+      }
+
+      var heightAttr = this.attr("height");
+      if (heightAttr === undefined) {
+        attrToProjector["height"] = (d, i, dataset) => {
+          var isTotal = totalAccessor(d, i, dataset);
+          var currentValue = this.y().accessor(d, i, dataset);
+          if (isTotal) {
+            return Math.abs(yScale.scale(<any> currentValue) - yScale.scale(<any> 0));
```

Comment:
One way to address this is by extending Bar<X, number> instead of Bar<X, Y>. If Plots.Waterfall only support the vertical orientation this is OK.